internal/servers/controller/handlers/targets/target_service_test.go

```diff
@@ -1264,3 +1266,479 @@ func TestRemoveTargetHostSets(t *testing.T) {
+			resultLibraryIds: []string{cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add library on populated target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "populated", target.WithCredentialLibraries([]string{cls[0].GetPublicId()})),
+			addLibraryIds:    []string{cls[1].GetPublicId()},
+			resultLibraryIds: []string{cls[0].GetPublicId(), cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add duplicated libraries on populated target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "duplicated", target.WithCredentialLibraries([]string{cls[0].GetPublicId()})),
+			addLibraryIds:    []string{cls[1].GetPublicId(), cls[1].GetPublicId()},
+			resultLibraryIds: []string{cls[0].GetPublicId(), cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add library on empty target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "empty for libraries"),
+			addLibraries:     []*pbs.CredentialLibrary{{Id: cls[1].GetPublicId()}},
+			resultLibraryIds: []string{cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add library on populated target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "populated for libraries", target.WithCredentialLibraries([]string{cls[0].GetPublicId()})),
+			addLibraries:     []*pbs.CredentialLibrary{{Id: cls[1].GetPublicId()}},
+			resultLibraryIds: []string{cls[0].GetPublicId(), cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add duplicated libraries on populated target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "duplicated for libraries", target.WithCredentialLibraries([]string{cls[0].GetPublicId()})),
+			addLibraries:     []*pbs.CredentialLibrary{{Id: cls[1].GetPublicId()}, {Id: cls[1].GetPublicId()}},
+			resultLibraryIds: []string{cls[0].GetPublicId(), cls[1].GetPublicId()},
+		},
+		{
+			name:             "Add duplicated libraries across 2 fields on populated target",
+			tar:              target.TestTcpTarget(t, conn, proj.GetPublicId(), "duplicated mixed", target.WithCredentialLibraries([]string{cls[0].GetPublicId()})),
+			addLibraryIds:    []string{cls[1].GetPublicId()},
+			addLibraries:     []*pbs.CredentialLibrary{{Id: cls[1].GetPublicId()}},
+			resultLibraryIds: []string{cls[0].GetPublicId(), cls[1].GetPublicId()},
+		},
+	}
+
+	for _, tc := range addCases {
+		t.Run(tc.name, func(t *testing.T) {
+			req := &pbs.AddTargetCredentialLibrariesRequest{
+				Id:                   tc.tar.GetPublicId(),
+				Version:              tc.tar.GetVersion(),
+				CredentialLibraryIds: tc.addLibraryIds,
+				CredentialLibraries:  tc.addLibraries,
+			}
+
+			got, err := s.AddTargetCredentialLibraries(auth.DisabledAuthTestContext(iamRepoFn, proj.GetPublicId()), req)
+			require.NoError(t, err, "Got error: %v", s)
+
+			assert.ElementsMatch(t, tc.resultLibraryIds, got.GetItem().GetCredentialLibraryIds())
+
+			if len(tc.resultLibraries) != 0 {
```

Comment:
All test cases have an empty `resultLibraries`, so this if is never entered

Reply:
Thank you!